src/main/java/io/aiven/kafka/auth/AivenAclAuthorizer.java

```diff
@@ -79,42 +68,53 @@ public void configure(final java.util.Map<String, ?> configs) {
         logDenials = config.logDenials();
 
         configFile = config.getConfigFile();
-        jsonReader = new AclJsonReader(configFile.toPath());
-        checkAndUpdateConfig();
-    }
-
-    /**
-     * Read config file and populate ACL entries, if the config file has changed.
-     * This function assumes appropriate synchronization by caller.
-     */
-    private void checkAndUpdateConfig() {
-        final long configFileLastModified = configFile.lastModified();
-
-        if (configFileLastModified != lastModifiedTimestamp) {
-            LOGGER.info("Reloading ACL configuration {}", configFile);
-            try {
-                final List<AivenAcl> newAclEntries = jsonReader.read();
-                // initialize cache for non-trivial ACLs
-                if (newAclEntries.size() > 10) {
-                    if (verdictCache != null) {
-                        verdictCache.clear();
-                    } else {
-                        verdictCache = new ConcurrentHashMap<>();
-                    }
-                } else {
-                    verdictCache = null;
+        final AclJsonReader jsonReader = new AclJsonReader(configFile.toPath());
+        cacheReference.set(VerdictCache.create(loadAcls(jsonReader)));
+        final AtomicReference<WatchKey> watchKeyReference = new AtomicReference<>(subscribeToAclChanges(configFile));
+        scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            final WatchKey watchKey = watchKeyReference.get();
+            if (watchKey != null) {
+                final List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
+
+                watchEvents.stream().filter(watchEvent -> {
+                    @SuppressWarnings("unchecked")
+                    final Path path = ((WatchEvent<Path>) watchEvent).context();
+                    // If the file is deleted `ENTRY_MODIFY` is also triggered, so we need check that the file exists.
+                    return configFile.exists() && path.endsWith(configFile.getName());
+                }).findFirst().ifPresent(watchEvent -> {
+                    LOGGER.info(watchEvent.kind() + ": " + watchEvent.context()
+                            + ", Modified: " + configFile.lastModified());
```

Comment:
It's a good practice to not use concatenation in logging - easier to read, no unnecessary string concatenation if the actual logging is not performed, etc:
```suggestion
                    LOGGER.info("{}: {}, Modified: {}",
                        watchEvent.kind(), watchEvent.context(), configFile.lastModified());
```